src/main/java/seedu/bankwithus/user/TransactionList.java

```diff
@@ -75,12 +80,44 @@ public int getSize() {
         return size;
     }
 
+    /**
+     * Prints all transactions in the transaction list.
+     *
+     * @throws NoTransactionsFoundException thrown when there are no transactions in the transaction list
+     */
     public void printAllTransactions() throws NoTransactionsFoundException {
         if (size == 0) {
             throw new NoTransactionsFoundException();
         }
         for (int i = 0; i < size; i++) {
-            System.out.println(transactions.get(i).toString());
+            System.out.println(i + 1 + ". " + transactions.get(i).toString());
+        }
+    }
+
+    /**
+     * Deletes a transaction from the transaction list.
+     * @param args the index of the transaction to be deleted
+     */
+    public void deleteTransaction(String args) throws NoTransactionsFoundException,
+            NumberFormatException, IndexOutOfBoundsException {
+        Ui ui = new Ui();
+        if (size == 0) {
+            throw new NoTransactionsFoundException();
+        } else if (Integer.parseInt(args) > size || Integer.parseInt(args) < 1) {
```

Comment:
if the args here is not a valid integer, then `Integer.parseInt(args)` will already throw a NumberFormatException. So I think it might be good to consider how to refactor this without the try and catch in the else.

```
        if (size == 0) {
            throw new NoTransactionsFoundException();
        }
        try {
            int index = Integer.parseInt(args) - 1;
            if (index > size || index < 1) {
                throw new IndexOutOfBoundsException();
            }
            transactions.remove(index);
            size--;
            ui.showTransactionDeletedMessage();
        } catch (NumberFormatException e) {
            throw new NumberFormatException();
        } catch (IndexOutOfBoundsException e) {
            throw new IndexOutOfBoundsException();
        } catch (Exception e) {
            e.printStackTrace();
        }
```

I think something like the above might be a better way to refactor.

Reply:
Ok, thank you for the suggestion! I'll try and refactor the code.

Reply:
Since this PR was closed, I'll try improving the code tomorrow by opening a new PR.